h5grove/content.py

```diff
@@ -218,14 +227,13 @@ def _get_child_metadata_content(self, depth=0):
             for child_path in self._h5py_entity.keys()
         ]
 
-    def metadata(self, depth: int = 1):
+    def metadata(self, depth: int = 1) -> GroupMetadata:
         """Metadata of the group. Recursively includes child metadata if depth > 0.
 
         :parameter depth: The level of child metadata resolution.
-        :returns: {"attributes": AttributeMetadata, "children": ChildMetadata, "name": str, "kind": str}
         """
         if depth <= 0:
-            return super().metadata()
+            return cast(GroupMetadata, super().metadata())
```

Comment:
Had to help mypy here to force inferring the return type as `GroupMetadata` instead of `ResolvedEntityMetadata`. Also, I just made `children` optional in `GroupMetatada`, assuming that there is no way to have a conditional return type based on `depth`.